boards/myboard/myfc/src/CMakeLists.txt

```diff
@@ -0,0 +1,17 @@
+add_library(drivers_board
+	# i2c.cpp
```

Comment:
I see 12c, serial and spin in NuttX https://github.com/apache/incubator-nuttx/tree/master/arch/arm/src/rp2040

Assuming they are not stubbed out there, these can came in 

I2c in nuttx already has reference counting to that is good to go! 
Tad bit of caution the CONFIG_I2C_RESET is not really supported (it is currently a nop in nuttx)